Niewim19/mstone1 integr

niewim19 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes shortly -->
Integration tests changes. Devided in atomic commits. Changes regarding:
 - using `motlin` in `mma`
 - changes in pid
 - adding height plot to quadLIve
 - shortening the logging messages and printing them only once for 20 control loops (*this is sketchy*)
 - changes in ekf barometer update step

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

<!--- In case of breaking change - please advice here what needs to be done in dependent projects. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

HBuczynski commented on quadcontrol/control.h:
```diff
@@ -26,7 +26,8 @@ typedef enum { flight_takeoff = 0, flight_pos, flight_hover, flight_landing, fli
 typedef struct {
-	int32_t alt; /* altitude in 1E-3 [m] (millimetres) above MSL */
+	int32_t alt;     /* altitude in 1E-3 [m] (millimetres) above MSL */
+	float spoolTime; /* time of engine spool up in milliseconds */
```
My suggestion to use shorter names.

```suggestion
	float time; /* time of motors spool up in milliseconds */
```

If you decide to change this variable, please update `static int quad_takeoff(const flight_mode_t *mode)` as well.

HBuczynski commented on ekf/ekflib.c:
```diff
@@ -94,24 +94,32 @@ static time_t ekf_dtGet(void)
+
 	while (ekf_common.run == 1) {
-		usleep(1000);
+		usleep(300);
```
The resolution in Phoenix-RTOS for `zynq-7000` target is 1 ms. This change will not have an effect. Sleep will be the same, 1 ms.

niewim19 replied:
I did not knew that. Thanks!

HBuczynski commented on quadcontrol/control.c:
```diff
@@ -167,13 +172,11 @@ static int quad_takeoff(const flight_mode_t *mode)
 		if (quad_motorsCtrl(throttle, mode->takeoff.alt, 0, 0, ANGLE_HOLD) < 0) {
 			return -1;
 		}
-		usleep(1000 * 100);
```
As I remember correctly, this delay was needed to have a smooth takeoff. Won't it have a negative impact on takeoff?
What is the relation of this change to logging?

niewim19 replied:
Sleep was used when _step-by-step_ spool up approach was used. Now it is more linear as the throttle is calculated in linear time space using the `elapsedTime / totalTime` coefficient, not by increasing it in steps.

HBuczynski commented on quadcontrol/control.c:
```diff
@@ -174,18 +186,29 @@ static int quad_takeoff(const flight_mode_t *mode)
 #if TEST_ATTITUDE
-	while (quad_timeMsGet() < now + quad_common.duration) {
+	while (now < end) {
+		if (now - lastLog > LOG_PERIOD) {
+			lastLog = now;
+			log_enable();
+		}
```
To have `enable/disable` in one place;
```suggestion
               now = quad_timeMsGet();
		if (now - lastLog > LOG_PERIOD) {
			lastLog = now;
			log_enable();
		}
		else {
		        log_disable();
		}
```

niewim19 commented:
- Implemented review remark about enable/disable in one place
 - Added 2 new commits at the end of tree about
   - refactoring `takeoff` mode
   - refactoring `landing` mode

niewim19 commented:
- Implemented review remark about enable/disable in one place
 - Added 2 new commits at the end of tree about
   - refactoring `takeoff` mode
   - refactoring `landing` mode

niewim19 commented on quadcontrol/control.c:
```diff
@@ -154,16 +159,32 @@ static int quad_motorsCtrl(float throttle, int32_t alt, int32_t roll, int32_t pi
+		if (now - lastLog > LOG_PERIOD) {
+			lastLog = now;
+			log_enable();
+		} else {
```
Resolved in next commit

niewim19 commented:
Amended last commit regarding landing procedure with:
 - addition of `flight_landing_t` structure addition
 - using `flight_landing_t` field `time` to determine spool down time in landing
